GSA-TTS/FAC
# New management command - Manually move "submitted" report to dissemination

Addresses part 2 of #4272 - providing an administrative command to move a SAC to the `disseminated` state.

The command is: `python manage.py move_to_disseminated --report_id ID_GOES_HERE`. This will look for a SAC with the `report_id` that you entered, and attempt to move it to the `disseminated` status - ONLY if it is stuck as `submitted` and passes cross-validation with no errors.

In the event that dissemination fails, or cross-validation fails, the SAC is returned to the `auditee_certified` state.

Additionally, this PR brings in some curation work from `jadudm/curation-api`. An app for `curation` has been created, which builts SQL scripts for tracking the before/after state of an audit. The above management command will track the audit's state before and after dissemination if it is successful.

## How to test

Before attempting the steps below, make sure you do a `docker-clean` and `docker-first-run`.

1. Create a SAC and bring it to the disseminated state.
2. Using a RDBMS, find this SAC, change `submission_status` to `submitted`, and ADD a row to `transition_name` and `transition_date`, which should be `disseminated` and a date for this action respectively.
3. Truncate the `dissemination_general` table to remove all dissemination data of this SAC. This will allow the management command to attempt to redisseminate without any issue.
4. Run the management command against your SAC.
5. Verify that the `fac_accepted_date` of the disseminated record matches the FIRST occurrence that the SAC was submitted (There should be two instances of `submitted` in `transition_name` and `transition_date` at this point).
6. Verify there is a record of this change in the `curation.record_version` table.

## PR Checklist: Submitter


## PR Checklist: Reviewer


The larger the PR, the stricter we should be about these points.

## Pre Merge Checklist: Merger
